analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/LocksReleasedAllPathsBase.cs

```diff
@@ -49,11 +49,37 @@ public abstract class LocksReleasedAllPathsBase : SymbolicRuleCheck
 
         // ToDo: Implement early bail-out if there's no interesting descendant node in context.Node to avoid useless SE runs
         public override bool ShouldExecute() =>
-            NodeContext.Node.DescendantNodes().OfType<IdentifierNameSyntax>().Any(x => x.Identifier.Text.Contains("Exit") || x.Identifier.Text.Contains("ReleaseReaderLock"));
+            NodeContext.Node.DescendantNodes().OfType<IdentifierNameSyntax>().Any(x => x.Identifier.Text.Contains("Exit")
+                                                                                       || x.Identifier.Text.Contains("ReleaseMutex")
+                                                                                       || x.Identifier.Text.Contains("ReleaseReaderLock"));
 
         public override ProgramState PostProcess(SymbolicContext context)
         {
-            if (context.Operation.Instance.AsInvocation() is { } invocation)
+            if (context.Operation.Instance.AsObjectCreation() is { } objectCreation)
+            {
+                if (objectCreation.Type.Is(KnownType.System_Threading_Mutex)
+                    && objectCreation.Arguments.FirstOrDefault() is { } firstArgument
+                    && firstArgument.AsArgument() is { } initiallyOwnedArgument
+                    && initiallyOwnedArgument.Value.AsLiteral() is { } literalOperation
+                    && literalOperation.WrappedOperation.ConstantValue.Value is true
```

Comment:
We can simplify this, because:
* `firstArgument` will always be an instance of an argument. We don't need to test it (not coverable)
* We don't care about the `Literal`. As long as `ConstantValue` is true, we don't care how it was produced (`0==0`, `Zero == 0`)
```suggestion
                    && firstArgument.AsArgument().Value.WrappedOperation.ConstantValue.Value is true
```

Reply:
Unfortunately it doesn't work exactly like that.
`firstArgument.AsArgument().Value.WrappedOperation.ConstantValue.Value` is false for the cases we need. 
What we actually need to check is `firstArgument.AsArgument().Value.WrappedOperation.Value.ConstantValue.Value` which is not accessible directly,

Reply:
Why not `objectCreation.Arguments.First().AsArgument()`? Building those extensions on the fly is handy.

Reply:
What I'd like is
```
&& objectCreation.Arguments.First().AsArgument().Value.Value.ConstantValue.Value is true)
```
The 1st `.Value` is due to nullability. The second `.Value` gives us the operation we need.